planetary-social/nos
# Add mention button to note composer

## Issues covered
#1561

## Description
This PR adds a mention button to the Note Composer. I couldn't find a non-complicated way to allow the button press mimic a user typing into a textView, so I kept it simple. It contains the following changes:
- Fixes a swuiftlint warning about a redundant type declaration in `NoteEditorController`.
- Adds a new @ image asset.
- Edits the append text function in `NoteEditorController` to include a check for the "@" text being appended, and showing the autoComplete list if that is true.
- Adds a mentionView to the `ComposerActionBar` that appends the "@" text and shows the autoComplete list, on press.

## How to test
1. Build the code.
2. Tap post button at the bottom of the screen.
3. Tap the @ button
4. Please check that the "@" text is visible on the editor and the autoComplete list shows up every time that button is pressed.

## Screenshots/Video

Before:
<img src="https://github.com/user-attachments/assets/2d8b7142-a133-489b-8f49-ed6333dce734" alt="Before" width="250"/>

After:
https://github.com/user-attachments/assets/ff06778d-dd62-4baf-966f-34143af7ec3d
